server.go

```diff
@@ -383,6 +384,12 @@ func writeErr(c ssh.Channel, id uint32, err error) error {
 		code = ssh_FX_OK
 	case io.EOF:
 		code = ssh_FX_EOF
+	default:
+		if os.IsPermission(err) {
+			code = ssh_FX_PERMISSION_DENIED
+		} else if os.IsNotExist(err) {
+			code = ssh_FX_NO_SUCH_FILE
+		}
 	}
```

Comment:
You could consider taking the nested if/else if statement and merge it with the existing switch cases, like this:

```Go
switch err {
case err == nil:
	code = ssh_FX_OK
case err == io.EOF:
	code = ssh_FX_EOF
case os.IsPermission(err):
	code = ssh_FX_PERMISSION_DENIED
case os.IsNotExist(err):
	code = ssh_FX_NO_SUCH_FILE
}
```

You could even move the `ssh_FX_FAILURE` default value into the `default:` case, so all the logic for selecting code is in one place.